### PR TITLE
InstallLiberty will re-run if config properties have been changed.

### LINKED ISSUE #221: installLiberty task reports UP-TO-DATE when liberty version changes
Recently I updated a project to use the latest beta:

```js
    liberty {
        install {
            // use 1 liberty install for the whole repo
            baseDir = rootProject.buildDir
            runtimeUrl = "https://public.dhe.ibm.com/ibmdl/export/pub/software/websphere/wasdev/downloads/wlp/beta/wlp-beta-2018.2.0.0.zip"
        }
    }
```

And got a failure when features present in the new beta could not be found by the runtime or installed by installUtility.  Turns out this is because the `installLiberty` task is considered to be up-to-date if `liberty.install.baseDir` exists, but it does not check to make sure it's the right version of liberty installed.

The `libertyInstall` task should be updated to compare product versions of the maven/runtimeUrl dependency to the product version in `${liberty.install.baseDir}/lib/versions/WebSphereApplicationServer.properties` to ensure the runtime itself is up to date.

### PR DESCRIPTION
**Pull request by jjvilleg**
Fixes #221

### CODE REVIEW

File: src/main/groovy/io/openliberty/tools/gradle/tasks/InstallLibertyTask.groovy [hunk 1]
```diff
@@ -59,25 +59,57 @@ class InstallLibertyTask extends AbstractTask {
         return project.configurations.libertyRuntime
     }
 
+    @Input
+    @Optional
+    String getLibertyRuntimeUrl() {
+        return project.liberty.install.runtimeUrl
+    }
```
Comment (cherylking) [inline src/main/groovy/io/openliberty/tools/gradle/tasks/InstallLibertyTask.groovy:66]: What about the project.liberty.runtime properties?
Screen Shot 2021-02-26 at 3 17 17 PM

Comment (cherylking) [reply src/main/groovy/io/openliberty/tools/gradle/tasks/InstallLibertyTask.groovy:66]: That is in the liberty extension. We also support gradle properties for those.
Screen Shot 2021-02-26 at 3 21 01 PM

File: src/main/groovy/io/openliberty/tools/gradle/tasks/InstallLibertyTask.groovy [hunk 2]
```diff
@@ -59,25 +59,57 @@ class InstallLibertyTask extends AbstractTask {
...
-            String licenseFilePath = project.configurations.getByName('libertyLicense').getAsPath()
-            if (licenseFilePath) {
-                def command = "java -jar " + licenseFilePath + " --acceptLicense " + project.buildDir
-                def process = command.execute()
-                process.waitFor()
-            }
-            
-        } else {
-            logger.info ("Liberty is already installed at: " + getInstallDir(project))
+        def params = buildInstallLibertyMap(project)
```
Comment (cherylking) [inline src/main/groovy/io/openliberty/tools/gradle/tasks/InstallLibertyTask.groovy:102]: So why don't we need some check here? This won't get called if it was already installed?

**Review by cherylking — CHANGES_REQUESTED** [review]

File: src/main/groovy/io/openliberty/tools/gradle/tasks/InstallLibertyTask.groovy (hunk 2 above)
Comment (jjvilleg) [reply src/main/groovy/io/openliberty/tools/gradle/tasks/InstallLibertyTask.groovy:102]: Exactly. Even if the Inputs weren't up to date, this check would keep it from reinstalling WLP with the updated configuration. We are already adding a checking if the contents are there in this line
https://github.com/OpenLiberty/ci.gradle/blob/master/src/main/groovy/io/openliberty/tools/gradle/tasks/InstallLibertyTask.groovy#L51

**Review by cherylking — COMMENTED** [review]
So the changes look good, but we probably need a couple of tests to prove it works correctly.

**Comment by cherylking** [comment]
Looks like your changes are causing this test to fail (invalid installDir location):
https://github.com/OpenLiberty/ci.gradle/pull/588/checks?check_run_id=2000509793#step:9:10984

**Comment by cherylking** [comment]
And another test failure: https://github.com/OpenLiberty/ci.gradle/pull/588/checks?check_run_id=2000509793#step:9:19127

**Review by cherylking — APPROVED** [review]

**Pull request merged**